src/api/app/assets/javascripts/webui2/kiwi_editor.js

```diff
@@ -0,0 +1,400 @@
+        repoField.html('');
+        var foundoptions = false;
+        $.each(data, function (idx, val) {
+          repoField.append(new Option(val));
+          repoField.prop('disabled', false);
+          foundoptions = true;
+        });
+        if (!foundoptions)
+          repoField.append(new Option('No repos found'));
+      },
+      complete: function () {
+        $('.ui-autocomplete-loading').addClass('d-none');
+        repoField.trigger("change");
+      }
+  });
+}
+
+function kiwiRepositoriesSetupAutocomplete(fields) {
+  var projectField = fields.find('[name=target_project]');
+  var repoField = fields.find('[name=target_repo]');
+  var aliasField = fields.find('[name=alias_for_repo]');
+
+  projectField.autocomplete({
+    source: projectField.data('ajaxurl'),
+    minLength: 2,
+    select: function (event, ui) {
+      autocompleteKiwiRepositories(ui.item.value, repoField);
+    },
+    search: function() {
+      $(this).addClass('loading-spinner');
+    },
+    response: function() {
+      $(this).removeClass('loading-spinner');
+    }
+  });
+
+  projectField.change(function () {
+    autocompleteKiwiRepositories(projectField.val(), repoField);
+  });
+
+  repoField.change(function () {
+    var repoFieldValue = repoField.val();
+    if (repoField.val() === 'No repos found') {
+      repoFieldValue = '';
+    }
+    var sourcePath = fields.find("[id$='source_path']");
+    sourcePath.val("obs://" + projectField.val() + '/' + repoField.val());
+    aliasField.val(repoFieldValue + '@' + projectField.val()).
+      trigger("change");
+    var repoTypeField = fields.find("[id$='repo_type']");
+    repoTypeField.val('rpm-md');
+  });
+
+  aliasField.change(function () {
+    fields.find("[id$='alias']").val($(this).val());
+  });
+}
+
+function kiwiPackagesSetupAutocomplete(fields) {
+  var packageField = fields.find('[id$=name]');
+
+  packageField.autocomplete({
+    source: function (request, response) {
+      var repositories = $("#kiwi-repositories-list [id$='source_path']").map(function() { return this.value;}).get();
+      var repositoryDestroyFields = $("#kiwi-repositories-list [id$='_destroy']");
+      var usingProjectRepositories =  $('#kiwi_image_use_project_repositories').prop('checked');
+      repositoryDestroyFields.each(function(index, input){ // remove destroyed repositories from the payload
+        if ($(input).val() === "1") {
+          repositories.splice(index, 1);
+        }
+      });
+      var payload = { term: request.term, repositories: repositories };
+      if (usingProjectRepositories) {
+        payload.useProjectRepositories = true;
+      }
+      $.getJSON(packageField.data('ajaxurl'), payload,
+        function (data) {
+          response(data);
+        });
+    },
+    search: function() {
+      $(this).addClass('loading-spinner');
+    },
+    response: function() {
+      $(this).removeClass('loading-spinner');
+    },
+    minLength: 2
+  });
+}
+
+$(document).ready(function(){
```

Comment:
this is the global ready function you're modifiying - called on all routes. The test suite isn't running due to haml lint, but it would explode fantastically

Reply:
this comment from @coolo is important. You could wrap this in a function here and load it in the view file as:

```ruby
= content_for :ready_function do
  function_name();